modules/data.atmosphere/R/extract.nc.module.R

```diff
@@ -0,0 +1,45 @@
+.extract.nc.module <- function(cf.id, register, dir, met, str_ns, site, new.site, con, start_date, end_date, host, overwrite=FALSE) {
+  logger.info("Begin Standardize Met")
+  
+  if(register$scale=="regional"){ #### Site extraction
+    
+    logger.info("Site Extraction")
+    
+    input.id   <- cf.id[1]
+    outfolder  <- ifelse(host$name == "localhost", 
+                         file.path(dir, paste0(met,"_CF_site_",str_ns)),
+                         file.path(host$dbfiles, paste0(met,"_CF_site_",str_ns)))
+    pkg        <- "PEcAn.data.atmosphere"
+    fcn        <- "extract.nc"
+    formatname <- 'CF Meteorology'
+    mimetype   <- 'application/x-netcdf'
+    
+    ready.id <- convert.input(
+      input.id, outfolder, formatname, mimetype, site.id=site$id,
+      start_date, end_date, pkg, fcn, con=con, host=host, browndog=NULL,
+      write=TRUE, slat=new.site$lat, slon=new.site$lon, newsite=new.site$id,
+      overwrite=overwrite)
+    
+  } else if(register$scale=="site") { ##### Site Level Processing
+    logger.info("Gapfilling") # Does NOT take place on browndog!
```

Comment:
Well, typically regional products have typically been gapfilled already and site-level data typically doesn't have to be extracted.

More generally, if we go down the route I proposed then we'll move toward building the workflow more dynamically rather than either/or